aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -393,39 +412,103 @@ mod test {
         let mut aggregator_data = AggregatorData::default();
         aggregator_data.create_new_aggregator(aggregator_id_for_test(200), 200);
 
-        let aggregator = aggregator_data.get_aggregator(aggregator_id_for_test(200), 200);
+        let aggregator =
+            aggregator_data.get_aggregator(aggregator_id_for_test(200), 200, &*TEST_RESOLVER, true);
         assert_ok!(aggregator.add(100));
         assert_ok!(aggregator.read_and_materialize(&*TEST_RESOLVER, &aggregator_id_for_test(200)));
         assert_eq!(aggregator.value, 100);
     }
 
+    #[test]
+    fn test_materialize_known_aggregator_disabled() {
+        let mut aggregator_data = AggregatorData::default();
+        aggregator_data.create_new_aggregator(aggregator_id_for_test(200), 200);
+        let aggregator =
+            aggregator_data.get_aggregator(aggregator_id_for_test(200), 200, &*TEST_RESOLVER, true);
+        assert_ok!(aggregator.add(100));
+        assert_ok!(aggregator.read_and_materialize(&*TEST_RESOLVER, &aggregator_id_for_test(200)));
```

Comment:
This read materializes aggregator before you call `get_aggregator` in the test, so testing is not really hapenning. Let's just remove this line? :)

Reply:
I'm not sure I completely understand. I initially created an aggregator with `aggregator_enabled` set to true. I then added 100 and materialized the aggregator. I later get the same aggregator with `aggregator_enabled` flag set to false, and do some checks on it. Am I doing wrong?

Reply:
Sorry I was just confused with the purpose of this test :)